vscode-trace-common/src/types/customization.ts

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+import { QuickPickItem } from 'vscode';
+
+/**
+ * Top level object as it comes in from the server
+ */
+export interface CustomizationConfigObject {
+    id: string;
+    name: string;
+    description: string;
+    schema: Schema;
+}
+
+/**
+ * Top level schema with id and name
+ */
+export interface Schema {
+    $schema: string;
+    $id: string;
+    type?: string;
+    title?: string;
+    description: string;
+    default?: any;
+    items?: { type: string };
+    oneOf?: { required: string[] }[];
+    properties?: { [key: string]: SchemaProperty };
+    required?: string[];
+    additionalProperties?: boolean;
+}
+
+/**
+ * Type for schema objects (and nested schema objects)
+ */
+export interface SchemaProperty {
+    type?: string;
+    title?: string;
+    description: string;
+    default?: any;
+    items?: { type: string };
+    oneOf?: { required: string[] }[];
+    properties?: { [key: string]: SchemaProperty };
+    required?: string[];
+    additionalProperties?: boolean;
+    const?: any;
+    errorMessage?: any;
+}
+
+/**
+ * Types of values that can be used as defaults in schema
+ */
+export type DefaultValue = string | number | boolean | null | Record<string, any> | any[];
+
+/**
+ * Result of validating a JSON file against a schema
+ */
+export interface ValidationResult {
+    isValid: boolean;
+    content?: CustomizationSubmission;
+    errors?: string[];
+}
+
+export interface CustomizationSubmission {
```

Comment:
In order for configuration .json files and/or their content to be interchangeable between this JSON configuration editor and the python client, it would be better if 'id' was renamed to 'sourceTypeId' and 'options' was renamed to 'parameters'.